subsys/bluetooth/mesh/shell/shell.c

```diff
@@ -400,28 +491,30 @@ static int cmd_uuid(const struct shell *sh, size_t argc, char *argv[])
 	uint8_t uuid[16];
 	size_t len;
 
+	const struct bt_mesh_prov *prov = bt_mesh_prov_get();
+
+	if (argc < 2) {
+		char uuid_hex_str[32 + 1];
+
+		bin2hex(prov->uuid, 16, uuid_hex_str, sizeof(uuid_hex_str));
+
+		shell_print_ctx("Device UUID: %s", uuid_hex_str);
+		return 0;
+	}
+
 	len = hex2bin(argv[1], strlen(argv[1]), uuid, sizeof(uuid));
 	if (len < 1) {
 		return -EINVAL;
 	}
 
-	memcpy(dev_uuid, uuid, len);
-	(void)memset(dev_uuid + len, 0, sizeof(dev_uuid) - len);
+	(void)memset(uuid + len, 0, sizeof(uuid) - len);
+	memcpy((uint8_t *)prov->uuid, uuid, sizeof(uuid));
```

Comment:
The same as for `static_val`. It doesn't seem to be legal because uuid defined as const-qualifier.